src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs

```diff
@@ -295,7 +298,7 @@ protected override void VisitFinallyBlock(BoundStatement finallyBlock, ref Local
 
         protected override void VisitLabel(BoundLabeledStatement node)
         {
-            _labelsDefined.Add(node.Label);
+            _labelsDefined[node.Label] = _currentBlock;
```

Comment:
>_labelsDefined[node.Label] = _currentBlock [](start = 12, length = 42)

Minor point: Consider using `_labelsDefined.Add(node.Label, _currentBlock)` so an exception is thrown in the unexpected case of a duplicated label.

Reply:
So, I did this, but the bootstrap compiler was crashing due to a duplicate label :/

which I confess I didn't actually understand how...